### ISSUE TITLE
Attempted to load class "Redis" from namespace "ContainerMlRinL0" with Symfony 4.4

### ISSUE THREAD

**Issue by mbabker** [snc/SncRedisBundle#548]
(Note, I'm not entirely sure if I've stumbled upon a Symfony bug, a bug in the proxy manager, or a bug in this bundle, so I'm starting here)

Working on updating an application to Symfony 4.4, my `snc_redis.default` service is being dumped out as a virtual proxy and is generating the error message in the title.

This is the service definition being dumped into the container in my local dev environment:

```php
/**
 * Gets the private 'snc_redis.default' shared service.
 *
 * @return \Redis
 */
protected function getSncRedis_DefaultService($lazyLoad = true)
{
    if ($lazyLoad) {
        return $this->privates['snc_redis.default'] = $this->createProxy('Redis_c535e97', function () {
            return \Redis_c535e97::staticProxyConstructor(function (&$wrappedInstance, \ProxyManager\Proxy\LazyLoadingInterface $proxy) {
                $wrappedInstance = $this->getSncRedis_DefaultService(false);

                $proxy->setProxyInitializer(null);

                return true;
            });
        });
    }

    include_once \dirname(__DIR__, 4).'/vendor/snc/redis-bundle/Factory/PhpredisClientFactory.php';

    return (new \Snc\RedisBundle\Factory\PhpredisClientFactory(($this->privates['snc_redis.logger'] ?? $this->getSncRedis_LoggerService())))->create('\\Redis', [0 => $this->getEnv('REDIS_DSN')], ['connection_async' => false, 'connection_persistent' => false, 'connection_timeout' => 5, 'read_write_timeout' => NULL, 'iterable_multibulk' => false, 'throw_errors' => true, 'serialization' => 'default', 'profile' => 'default', 'cluster' => NULL, 'prefix' => NULL, 'service' => NULL], 'default');
}
```

And the relevant proxy it's referencing:

```php
class Redis_c535e97 extends Redis implements \ProxyManager\Proxy\VirtualProxyInterface
{
    // proxy class
}
```

I'm not sure why it's trying to generate a proxy for the `\Redis` class, but the proxy is being dumped out with an incorrect signature as it's missing the leading backslash to mark it as a global class.  Changing the default value of the phpredis_client class config to include the leading backslash doesn't make a difference here either.  Removing the `setLazy()` call in the extension class when the definition is being built does get rid of the issue, but I get the feeling that's not the right way to go about this.

**Comment by B-Galati** [comment]
@mbabker Hello and thanks for reporting.

The laziness is enable to prevent connection to redis server when the client is not used. For example when warming up the cache.

I am not able to reproduce your error. Which bundle version are you using? Which proxy manager version? What's your bundle config?

**Comment by mbabker** [comment]
Relevant bits from `composer show`:

```sh
ocramius/proxy-manager                           2.2.3              A library providing utilities to generate, instantiate and generally operate with Object Proxies
snc/redis-bundle                                 3.2.1              A Redis bundle for Symfony
symfony/asset                                    v4.4.0             Symfony Asset Component
symfony/browser-kit                              v4.4.0             Symfony BrowserKit Component
symfony/cache                                    v4.4.0             Symfony Cache component with PSR-6, PSR-16, and tags
symfony/cache-contracts                          v2.0.0             Generic abstractions related to caching
symfony/class-loader                             v3.4.35            Symfony ClassLoader Component
symfony/config                                   v4.4.0             Symfony Config Component
symfony/console                                  v4.4.0             Symfony Console Component
symfony/css-selector                             v4.4.0             Symfony CssSelector Component
symfony/debug                                    v4.4.0             Symfony Debug Component
symfony/debug-bundle                             v4.4.0             Symfony DebugBundle
symfony/debug-pack                               v1.0.7             A debug pack for Symfony projects
symfony/dependency-injection                     v4.4.0             Symfony DependencyInjection Component
symfony/doctrine-bridge                          v4.4.0             Symfony Doctrine Bridge
symfony/dom-crawler                              v4.4.0             Symfony DomCrawler Component
symfony/dotenv                                   v4.4.0             Registers environment variables from a .env file
symfony/error-handler                            v4.4.0             Symfony ErrorHandler Component
symfony/event-dispatcher                         v4.4.0             Symfony EventDispatcher Component
symfony/event-dispatcher-contracts               v1.1.7             Generic abstractions related to dispatching event
symfony/expression-language                      v4.4.0             Symfony ExpressionLanguage Component
symfony/filesystem                               v4.4.0             Symfony Filesystem Component
symfony/finder                                   v4.4.0             Symfony Finder Component
symfony/flex                                     v1.4.8             Composer plugin for Symfony
symfony/form                                     v4.4.0             Symfony Form Component
symfony/framework-bundle                         v4.4.0             Symfony FrameworkBundle
symfony/http-foundation                          v4.4.0             Symfony HttpFoundation Component
symfony/http-kernel                              v4.4.0             Symfony HttpKernel Component
symfony/inflector                                v4.4.0             Symfony Inflector Component
symfony/intl                                     v4.4.0             A PHP replacement layer for the C intl extension that includes additional data from the ICU library.
symfony/mime                                     v4.4.0             A library to manipulate MIME messages
symfony/monolog-bridge                           v4.4.0             Symfony Monolog Bridge
symfony/monolog-bundle                           v3.5.0             Symfony MonologBundle
symfony/options-resolver                         v4.4.0             Symfony OptionsResolver Component
symfony/orm-pack                                 v1.0.7             A pack for the Doctrine ORM
symfony/phpunit-bridge                           v4.4.0             Symfony PHPUnit Bridge
symfony/polyfill-ctype                           v1.12.0            Symfony polyfill for ctype functions
symfony/polyfill-intl-icu                        v1.12.0            Symfony polyfill for intl's ICU-related data and classes
symfony/polyfill-intl-idn                        v1.12.0            Symfony polyfill for intl's idn_to_ascii and idn_to_utf8 functions
symfony/polyfill-mbstring                        v1.12.0            Symfony polyfill for the Mbstring extension
symfony/process                                  v4.4.0             Symfony Process Component
symfony/profiler-pack                            v1.0.4             A pack for the Symfony web profiler
symfony/property-access                          v4.4.0             Symfony PropertyAccess Component
symfony/property-info                            v4.4.0             Symfony Property Info Component
symfony/proxy-manager-bridge                     v4.4.0             Symfony ProxyManager Bridge
symfony/routing                                  v4.4.0             Symfony Routing Component
symfony/security-bundle                          v4.4.0             Symfony SecurityBundle
symfony/security-core                            v4.4.0             Symfony Security Component - Core Library
symfony/security-csrf                            v4.4.0             Symfony Security Component - CSRF Library
symfony/security-guard                           v4.4.0             Symfony Security Component - Guard
symfony/security-http                            v4.4.0             Symfony Security Component - HTTP Integration
symfony/serializer                               v4.4.0             Symfony Serializer Component
symfony/serializer-pack                          v1.0.2             A pack for the Symfony serializer
symfony/service-contracts                        v2.0.0             Generic abstractions related to writing services
symfony/stopwatch                                v4.4.0             Symfony Stopwatch Component
symfony/swiftmailer-bundle                       v3.4.0             Symfony SwiftmailerBundle
symfony/templating                               v4.4.0             Symfony Templating Component
symfony/translation                              v4.4.0             Symfony Translation Component
symfony/translation-contracts                    v2.0.0             Generic abstractions related to translation
symfony/twig-bridge                              v4.4.0             Symfony Twig Bridge
symfony/twig-bundle                              v4.4.0             Symfony TwigBundle
symfony/validator                                v4.4.0             Symfony Validator Component
symfony/var-dumper                               v4.4.0             Symfony mechanism for exploring and dumping PHP variables
symfony/var-exporter                             v4.4.0             A blend of var_export() + serialize() to turn any serializable data structure to plain PHP code
symfony/web-link                                 v4.4.0             Symfony WebLink Component
symfony/web-profiler-bundle                      v4.4.0             Symfony WebProfilerBundle
symfony/webpack-encore-bundle                    v1.7.1             Integration with your Symfony app & Webpack Encore!
symfony/yaml                                     v4.4.0             Symfony Yaml Component
```

Bundle config:

```yaml
snc_redis:
    clients:
        default:
            type: 'phpredis'
            alias: 'default'
            dsn: '%env(REDIS_DSN)%'
            logging: '%kernel.debug%'
    session:
        client: 'default'
        prefix: '%env(REDIS_PREFIX)%'
    swiftmailer:
        client: 'default'
        key: '%env(REDIS_PREFIX)%_swiftmailer'
```

PHP info:

```sh
$ php --ri redis | grep Version
Redis Version => 4.3.0

$ php -v
PHP 7.3.6 (cli) (built: Jun 17 2019 16:06:12) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.3.6, Copyright (c) 1998-2018 Zend Technologies
    with Xdebug v2.7.2, Copyright (c) 2002-2019, by Derick Rethans
    with Zend OPcache v7.3.6, Copyright (c) 1999-2018, by Zend Technologies
```

Flipping back to my main dev branch which is in the same state as the 4.4 upgrade branch except for being on Symfony 4.3.8 and compare the compiled container, my `snc_redis.default` service ends up being dumped in separate files and things seem to be working OK since the Redis proxy is in its own non-namespaced file instead of the same file as my `srcApp_KernelDevDebugContainer` which Symfony dumps in a namespace.

So that got me to thinking a bit, and I had updated my Kernel to match the updated recipes repo.  Undoing the changes from https://github.com/symfony/recipes/commit/998b214c6db221001e185a8fdf259ab98e2f6051#diff-a3ba7df3efe564987f3475d2e67f5d93 got the container to dump out correctly again.

So like I said in the beginning, I very well may have run into a bug with another package, but it seemed easiest to start here and work my way through the system.  At this point it sounds like I should move this over to the Symfony repo?

**Comment by B-Galati** [comment]
Thank you for your insightful comments, that's really appreciated :+1: 

>At this point it sounds like I should move this over to the Symfony repo?

Yes please! Could you link to this issue so that we can follow and help?

**Comment by kricha** [comment]
Have the same issue.

**Comment by mbabker** [comment]
Well, I have no idea

**Comment by B-Galati** [comment]
@mbabker @kricha Is it fixed in Symfony 4.4.2 ?
